Blocks/HTTP/test.cc

```diff
@@ -106,6 +107,74 @@ TEST(HTTPAPI, Register) {
   EXPECT_EQ(1u, HTTP(FLAGS_net_api_test_port).HandlersCount());
 }
 
+TEST(HTTPAPI, RegisterWithURLPathParams) {
+  using current::strings::Join;
+
+  HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
+  auto handler = [](Request r) { r(r.url.path + ':' + Join(r.url_path_params, ',')); };
+  HTTP(FLAGS_net_api_test_port).Register("/", handler, URLPathParams::Count::Any);
+  HTTP(FLAGS_net_api_test_port)
+      .Register("/user", handler, URLPathParams::Count::One | URLPathParams::Count::Two);
+  HTTP(FLAGS_net_api_test_port).Register("/user/a", handler, URLPathParams::Count::One);
+  HTTP(FLAGS_net_api_test_port).Register("/user/a/1", handler, URLPathParams::Count::None);
+  ASSERT_THROW(HTTP(FLAGS_net_api_test_port).Register("/", handler), HandlerAlreadyExistsException);
+  ASSERT_THROW(HTTP(FLAGS_net_api_test_port).Register("/user", handler), HandlerAlreadyExistsException);
+  ASSERT_THROW(HTTP(FLAGS_net_api_test_port).Register("/user/a", handler), HandlerAlreadyExistsException);
+  ASSERT_THROW(HTTP(FLAGS_net_api_test_port).Register("/user/a/1", handler), HandlerAlreadyExistsException);
+
+  const string base_url = Printf("http://localhost:%d", FLAGS_net_api_test_port);
+  {
+    const auto response = HTTP(GET(base_url + "/"));
+    EXPECT_EQ(200, static_cast<int>(response.code));
+    EXPECT_EQ("/:", response.body);
+  }
+  {
+    const auto response = HTTP(GET(base_url + "/foo"));
+    EXPECT_EQ(200, static_cast<int>(response.code));
+    EXPECT_EQ("/:foo", response.body);
+  }
+  {
+    const auto response = HTTP(GET(base_url + "/foo/bar"));
+    EXPECT_EQ(200, static_cast<int>(response.code));
+    EXPECT_EQ("/:foo,bar", response.body);
```

Comment:
Coming from node.js, it took me a while to understand this notation. Why not:

`r(r.url.path + ':' + Join(r.url_path_params, ','));` -> `r(r.url.path + " (" + Join(r.url_path_params, ',') + ')');` ?